src/python/pants/engine/README.md

```diff
@@ -3,8 +3,9 @@
 ## API
 
 The end user API for the engine is based on the registration of `@rule`s, which are functions
-or coroutines with statically declared inputs and outputs. A Pants (plugin) developer can write
-and install additional `@rule`s to extend the functionality of Pants.
+or coroutines with statically declared parameter types (also known as `Param` types) and return type
```

Comment:
As I read it the grouping suggested by 1 is: "coroutines with statically declared (parameter types and return type)" whereas in 2 it's "coroutines with (statically declared parameter types) and (a return type)". The latter grouping is not what we want the reader to hear. This may be in the ear of the beholder though.